datadog_checks_base/datadog_checks/base/checks/base.py

```diff
@@ -75,6 +76,7 @@
 ONE_PER_CONTEXT_METRIC_TYPES = [aggregator.GAUGE, aggregator.RATE, aggregator.MONOTONIC_COUNT]
 
 
+@traced_class
```

Comment:
Necessary?

Reply:
If we want to have traces on `AgentCheck` method yes https://www.python.org/dev/peps/pep-0487/#calling-the-hook-on-the-class-itself, even though i'm not sure about the additional value of having traces on `AgentCheck` methods